addons/web/static/src/core/tree_editor/condition_tree.js

```diff
@@ -639,12 +639,16 @@ function _expressionFromTree(tree, options, isRoot = false) {
         return formatAST(operator === "=" ? not(pathAST) : pathAST);
     }
 
-    if (pathAST.type === 5 && isX2Many(pathAST, options) && ["in", "not in"].includes(operator)) {
-        const valueAST = toAST(value);
-        const otherIteratorAST = [4, 10].includes(valueAST.type)
-            ? valueAST
-            : { type: 4, value: [valueAST] };
+    let valueAST = toAST(value);
+    if (
+        ["in", "not in"].includes(operator) &&
+        !(value instanceof Expression) &&
```

Comment:
!(value instanceof Expression) is the important part. The rest is ref/factorization